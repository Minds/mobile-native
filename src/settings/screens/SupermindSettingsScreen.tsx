import { showNotification } from 'AppMessages';
import { observer, useLocalStore } from 'mobx-react';
import React from 'react';
import { ScrollView } from 'react-native';
import CenteredLoading from '~/common/components/CenteredLoading';
import ErrorLoading from '~/common/components/ErrorLoading';
import InputContainer from '~/common/components/InputContainer';
import useApiFetch, { FetchStore } from '~/common/hooks/useApiFetch';
import apiService from '~/common/services/api.service';
import i18n from '~/common/services/i18n.service';
import mindsConfigService from '~/common/services/minds-config.service';

import {
  B1,
  B2,
  Button,
  Screen,
  ScreenHeader,
  ScreenSection,
} from '~/common/ui';
import StripeConnectButton from '../../wallet/v2/stripe-connect/StripeConnectButton';

type Settings = {
  min_offchain_tokens: number;
  min_cash: number;
};

export default observer(function SupermindSettingsScreen({ navigation }) {
  const fetchStore = useApiFetch<Settings>('api/v3/supermind/settings');
  const localStore = useLocalStore(createStore, { navigation, fetchStore });

  /**
   * Sync values with local store
   */
  React.useEffect(() => {
    if (fetchStore.result) {
      localStore.setCash(fetchStore.result.min_cash.toString());
      localStore.setTokens(fetchStore.result.min_offchain_tokens.toString());
      const config = mindsConfigService.getSettings();
      if (config.supermind?.min_thresholds) {
        localStore.setThresholds(config.supermind.min_thresholds);
      }
    }
  }, [fetchStore.result, localStore]);

  return (
    <Screen safe>
      <ScreenHeader
        title="Supermind"
        back
        extra={
          <Button
            mode="flat"
            type="action"
            testID="save"
            spinner
            disabled={Boolean(localStore.cashError || localStore.tokenError)}
            onPress={() => localStore.submit()}>
            {i18n.t('save')}
          </Button>
        }
      />
      <ScrollView>
        <ScreenSection>
          <B1 font="medium" color="tertiary" vertical="L">
            {i18n.t('supermind.getPaid')}
          </B1>
          <B1 vertical="M">{i18n.t('supermind.setMinimum')}</B1>
          <B2 bottom="M" color="secondary">
            {i18n.t('supermind.fansCant')}
          </B2>
        </ScreenSection>
        <Inputs fetchStore={fetchStore} store={localStore} />
        <StripeConnectButton top="M" bottom="L" />
      </ScrollView>
    </Screen>
  );
});

/**
 * Inputs (observer)
 */
const Inputs = observer(
  ({
    fetchStore,
    store,
  }: {
    fetchStore: FetchStore<Settings>;
    store: LocalStore;
  }) =>
    fetchStore.loading ? (
      <CenteredLoading />
    ) : fetchStore.result ? (
      <>
        <InputContainer
          testID="tokensInput"
          placeholder={i18n.t('tokens')}
          onChangeText={v => {
            store.setTokens(v);
          }}
          keyboardType="numeric"
          error={store.tokenError}
          value={store.tokens.toString()}
          noBottomBorder
          autoFocus
        />
        <InputContainer
          testID="cashInput"
          placeholder={i18n.t('usd')}
          keyboardType="numeric"
          error={store.cashError}
          onChangeText={v => {
            store.setCash(v);
          }}
          value={store.cash.toString()}
        />
      </>
    ) : (
      <ErrorLoading
        tryAgain={() => fetchStore.fetch()}
        message={i18n.t('errorMessage')}
      />
    ),
);

type Thresholds = { min_cash: number; min_offchain_tokens: number };

/**
 * Local store
 */
const createStore = ({ navigation }) => {
  return {
    tokens: '1',
    cash: '10',
    cashError: '',
    tokenError: '',
    min_thresholds: { min_cash: 10, min_offchain_tokens: 1 },
    setThresholds(thresholds: Thresholds) {
      this.min_thresholds = thresholds;
    },
    setCash(v: string) {
      this.cash = v;
      this.cashError = '';
      this.cashError = this.validateInput(v, this.min_thresholds.min_cash);
    },
    validateInput(value: string, minimum: number) {
      const numericValue = parseFloat(value) || 0;
      if (value === '') {
        return i18n.t('auth.fieldRequired');
      }
      if (numericValue < minimum) {
        return i18n.t('supermind.minimum', {
          value: minimum,
        });
      }
      if (value.includes('.') && value.split('.')[1].length > 2) {
        return i18n.t('supermind.maxTwoDecimals');
      }
      return '';
    },
    setTokens(v: string) {
      this.tokens = v;
      this.tokenError = '';
      this.tokenError = this.validateInput(
        v,
        this.min_thresholds.min_offchain_tokens,
      );
    },
    async submit() {
      try {
        await apiService.post('api/v3/supermind/settings', {
          min_cash: parseFloat(this.cash),
          min_offchain_tokens: parseFloat(this.tokens),
        });
      } catch (error) {
        if (error instanceof Error) {
          showNotification(error.message);
        } else {
          showNotification(i18n.t('error'));
        }
        return;
      }
      showNotification(i18n.t('settings.saved'));
      navigation.goBack();
      return [this.cash, this.tokens];
    },
  };
};

type LocalStore = ReturnType<typeof createStore>;
