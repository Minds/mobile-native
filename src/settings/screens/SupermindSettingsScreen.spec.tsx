import * as React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react-native';
import SupermindSettingsScreen from './SupermindSettingsScreen';
import apiService from '~/common/services/api.service';
import mindsConfigService from '~/common/services/minds-config.service';

jest.mock('~/common/hooks/use-stores');
jest.mock('~/common/services/permissions.service');
jest.mock('~/common/services/minds-config.service');
jest.mock('~/common/services/api.service');

const mockedApi = apiService as jest.Mocked<typeof apiService>;

const navigation = { goBack: jest.fn() };

describe('Supermind settings', () => {
  beforeEach(() => {
    jest.useRealTimers();
    mockedApi.get.mockClear();
    const resp: any = {
      min_cash: '12',
      min_offchain_tokens: '2',
    };
    // @ts-ignore
    mindsConfigService.getSettings.mockReturnValue({
      supermind: {
        min_thresholds: {
          min_cash: 10,
          min_offchain_tokens: 1,
        },
      },
    });

    mockedApi.get.mockResolvedValue(resp);
  });
  afterAll(() => {
    jest.useFakeTimers();
  });

  it('should load data and render correctly', async () => {
    render(<SupermindSettingsScreen navigation={navigation} />);

    // render the screen with the loading indicator
    expect(screen.toJSON()).toMatchSnapshot();

    await waitFor(() => screen.getAllByTestId('tokensInput'), {
      timeout: 15000,
      interval: 100,
    });

    // render the screen with data and the inputs should be shown
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('should update the store and submit', async () => {
    render(<SupermindSettingsScreen navigation={navigation} />);

    // render the screen with the loading indicator
    expect(screen.toJSON()).toMatchSnapshot();

    await waitFor(() => screen.getAllByTestId('tokensInput'), {
      timeout: 4000,
      interval: 100,
    });

    // fill form
    await fireEvent.changeText(screen.getByTestId('tokensInput'), '2');
    await fireEvent.changeText(screen.getByTestId('cashInput'), '12');

    // tap save
    fireEvent.press(screen.getByTestId('save'));

    // it should call the endpoint
    expect(mockedApi.post).toHaveBeenCalledWith('api/v3/supermind/settings', {
      min_cash: 12,
      min_offchain_tokens: 2,
    });
  });
});
