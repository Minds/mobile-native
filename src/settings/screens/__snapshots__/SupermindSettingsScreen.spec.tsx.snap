// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supermind settings should load data and render correctly 1`] = `
<View>
  <View
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginRight": 8,
              "position": "relative",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={9.333333333333332}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 28,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "width": 28,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#AEB0B8",
                        "fontSize": 28,
                      },
                      Object {},
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅁
                </Text>
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
              },
              Object {
                "fontFamily": "Roboto-Bold",
              },
              Object {
                "fontSize": 24,
                "lineHeight": 32,
              },
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          Supermind
        </Text>
      </View>
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
          testID="save"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 100,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "overflow": "hidden",
                },
                Object {
                  "borderColor": "transparent",
                  "borderRadius": 4,
                },
                Object {
                  "minHeight": 42,
                  "paddingHorizontal": 24,
                  "paddingVertical": 2,
                },
                undefined,
                Object {
                  "borderRadius": 0,
                  "minHeight": 32,
                  "paddingHorizontal": 8,
                  "paddingVertical": 4,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 100,
                  "borderWidth": 1,
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 0,
                      },
                    ],
                  }
                }
              >
                <ActivityIndicator
                  color="#FFFFFF"
                  size="small"
                />
              </View>
            </View>
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.05)",
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#1B85D6",
                    },
                    Object {
                      "fontFamily": "Roboto-Medium",
                    },
                    Object {
                      "fontSize": 18,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#797B82",
                },
                Object {
                  "fontFamily": "Roboto-Medium",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Get paid to reply to your fans.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginVertical": 12,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "fontFamily": "Roboto",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Set your minimum offer amount
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 12,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#B8C1CA",
                },
                Object {
                  "fontFamily": "Roboto",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Fans can’t send you Supermind offers for less than your minimum offer amount.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
            "minHeight": 100,
          }
        }
      >
        <ActivityIndicator
          color="#1B85D6"
          size="large"
        />
      </View>
      <ActivityIndicator
        color="#B8C1CA"
      />
    </View>
  </RCTScrollView>
</View>
`;

exports[`Supermind settings should load data and render correctly 2`] = `
<View>
  <View
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginRight": 8,
              "position": "relative",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={9.333333333333332}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 28,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "width": 28,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#AEB0B8",
                        "fontSize": 28,
                      },
                      Object {},
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅁
                </Text>
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
              },
              Object {
                "fontFamily": "Roboto-Bold",
              },
              Object {
                "fontSize": 24,
                "lineHeight": 32,
              },
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          Supermind
        </Text>
      </View>
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
          testID="save"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 100,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "overflow": "hidden",
                },
                Object {
                  "borderColor": "transparent",
                  "borderRadius": 4,
                },
                Object {
                  "minHeight": 42,
                  "paddingHorizontal": 24,
                  "paddingVertical": 2,
                },
                undefined,
                Object {
                  "borderRadius": 0,
                  "minHeight": 32,
                  "paddingHorizontal": 8,
                  "paddingVertical": 4,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 100,
                  "borderWidth": 1,
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 0,
                      },
                    ],
                  }
                }
              >
                <ActivityIndicator
                  color="#FFFFFF"
                  size="small"
                />
              </View>
            </View>
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.05)",
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#1B85D6",
                    },
                    Object {
                      "fontFamily": "Roboto-Medium",
                    },
                    Object {
                      "fontSize": 18,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#797B82",
                },
                Object {
                  "fontFamily": "Roboto-Medium",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Get paid to reply to your fans.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginVertical": 12,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "fontFamily": "Roboto",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Set your minimum offer amount
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 12,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#B8C1CA",
                },
                Object {
                  "fontFamily": "Roboto",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Fans can’t send you Supermind offers for less than your minimum offer amount.
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 16,
            },
            Object {
              "paddingTop": 8,
            },
            Object {
              "backgroundColor": "#1A2025",
            },
            Object {
              "borderTopWidth": 1,
            },
            null,
            Object {
              "borderColor": "#414C57",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 5,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "stretch",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                      },
                      Object {
                        "fontFamily": "Roboto",
                      },
                      Array [
                        Object {
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "marginBottom": 0,
                        },
                        Object {
                          "color": "#B8C1CA",
                        },
                        Array [
                          Object {
                            "color": "#B8C1CA",
                          },
                          Object {
                            "fontSize": 16,
                          },
                          Object {
                            "paddingLeft": 1,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  Tokens
                </Text>
              </View>
            </View>
            <TextInput
              autoCapitalize="none"
              autoFocus={true}
              error=""
              keyboardAppearance="dark"
              keyboardType="numeric"
              labelStyle={
                Array [
                  Object {
                    "color": "#B8C1CA",
                  },
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "paddingLeft": 1,
                  },
                  undefined,
                ]
              }
              onChangeText={[Function]}
              placeholderTextColor="#797B82"
              returnKeyType="done"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "#414C57",
                    "borderRadius": 2,
                    "borderWidth": 1,
                    "color": "#FFFFFF",
                    "fontFamily": "Roboto",
                    "fontSize": 16,
                    "lineHeight": 21,
                  },
                  Array [
                    Object {
                      "borderWidth": 0,
                      "marginBottom": 7,
                      "paddingHorizontal": 0,
                      "paddingVertical": 6,
                    },
                    Object {
                      "paddingLeft": 1,
                    },
                    Object {
                      "fontSize": 18,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                    undefined,
                  ],
                ]
              }
              testID="tokensInput"
              underlineColorAndroid="transparent"
              value="2"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 16,
            },
            Object {
              "paddingTop": 8,
            },
            Object {
              "backgroundColor": "#1A2025",
            },
            Object {
              "borderTopWidth": 1,
            },
            Object {
              "borderBottomWidth": 1,
            },
            Object {
              "borderColor": "#414C57",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 5,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "stretch",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                      },
                      Object {
                        "fontFamily": "Roboto",
                      },
                      Array [
                        Object {
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "marginBottom": 0,
                        },
                        Object {
                          "color": "#B8C1CA",
                        },
                        Array [
                          Object {
                            "color": "#B8C1CA",
                          },
                          Object {
                            "fontSize": 16,
                          },
                          Object {
                            "paddingLeft": 1,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  USD
                </Text>
              </View>
            </View>
            <TextInput
              autoCapitalize="none"
              error=""
              keyboardAppearance="dark"
              keyboardType="numeric"
              labelStyle={
                Array [
                  Object {
                    "color": "#B8C1CA",
                  },
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "paddingLeft": 1,
                  },
                  undefined,
                ]
              }
              onChangeText={[Function]}
              placeholderTextColor="#797B82"
              returnKeyType="done"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "#414C57",
                    "borderRadius": 2,
                    "borderWidth": 1,
                    "color": "#FFFFFF",
                    "fontFamily": "Roboto",
                    "fontSize": 16,
                    "lineHeight": 21,
                  },
                  Array [
                    Object {
                      "borderWidth": 0,
                      "marginBottom": 7,
                      "paddingHorizontal": 0,
                      "paddingVertical": 6,
                    },
                    Object {
                      "paddingLeft": 1,
                    },
                    Object {
                      "fontSize": 18,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                    undefined,
                  ],
                ]
              }
              testID="cashInput"
              underlineColorAndroid="transparent"
              value="12"
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Supermind settings should update the store and submit 1`] = `
<View>
  <View
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginRight": 8,
              "position": "relative",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={9.333333333333332}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 28,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "width": 28,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#AEB0B8",
                        "fontSize": 28,
                      },
                      Object {},
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅁
                </Text>
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
              },
              Object {
                "fontFamily": "Roboto-Bold",
              },
              Object {
                "fontSize": 24,
                "lineHeight": 32,
              },
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          Supermind
        </Text>
      </View>
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
            }
          }
          testID="save"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 100,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "overflow": "hidden",
                },
                Object {
                  "borderColor": "transparent",
                  "borderRadius": 4,
                },
                Object {
                  "minHeight": 42,
                  "paddingHorizontal": 24,
                  "paddingVertical": 2,
                },
                undefined,
                Object {
                  "borderRadius": 0,
                  "minHeight": 32,
                  "paddingHorizontal": 8,
                  "paddingVertical": 4,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 100,
                  "borderWidth": 1,
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 0,
                      },
                    ],
                  }
                }
              >
                <ActivityIndicator
                  color="#FFFFFF"
                  size="small"
                />
              </View>
            </View>
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.05)",
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#1B85D6",
                    },
                    Object {
                      "fontFamily": "Roboto-Medium",
                    },
                    Object {
                      "fontSize": 18,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 16,
          }
        }
      >
        <View
          style={
            Object {
              "marginVertical": 16,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#797B82",
                },
                Object {
                  "fontFamily": "Roboto-Medium",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Get paid to reply to your fans.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginVertical": 12,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "fontFamily": "Roboto",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Set your minimum offer amount
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 12,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#B8C1CA",
                },
                Object {
                  "fontFamily": "Roboto",
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Fans can’t send you Supermind offers for less than your minimum offer amount.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
            "minHeight": 100,
          }
        }
      >
        <ActivityIndicator
          color="#1B85D6"
          size="large"
        />
      </View>
      <ActivityIndicator
        color="#B8C1CA"
      />
    </View>
  </RCTScrollView>
</View>
`;
